packages/apps/shopify-app-remix/src/server/authenticate/admin/scope/request.ts

```diff
@@ -13,7 +20,16 @@ export function requestScopesFactory(params: BasicParams, session: Session) {
     });
 
     if (scopes.length === 0) return;
+    if (await alreadyGranted(scopes, admin)) return;
```

Comment:
@lizkenyon you are right, there's no problem in that case, the user will need to grant only the ones that are missing. The issue, and that's why we introduced this check before the request, is that if all the requested scopes are already granted then a loop detector in the server side is run, and this can lead the user to an error.